src/System.Net.Security/src/System/Net/Unix/SecuritySafeHandles.cs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Authentication;
+using System.Security.Authentication.ExtendedProtection;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Security
+{
+#if DEBUG
+    internal sealed class SafeFreeCertContext : DebugSafeHandle
+    {
+#else
+    internal sealed class SafeFreeCertContext : SafeHandle
+    {
+#endif
+        public SafeFreeCertContext(IntPtr handle) : base(handle, true)
+        {
+        }
+
+        public override bool IsInvalid
+        {
+            get
+            {
+                return IntPtr.Zero == handle;
+            }
+        }
+
+        protected override bool ReleaseHandle()
+        {
+            if (IntPtr.Zero != handle)
+            {
+                Interop.libcrypto.X509_free(handle);
+            }
+            return true;
+        }
+    }
+
+    //
+    // Implementation of handles dependable on FreeCredentialsHandle
+    //
+#if DEBUG
+    internal sealed class SafeFreeCredentials : DebugSafeHandle
+    {
+#else
+    internal sealed class SafeFreeCredentials : SafeHandle
+    {
+#endif
```

Comment:
I don't understand what's being done with SafeHandle here... SetHandle is never being called, so we're never actually storing a handle into this.  Seems like it's really just being used as a finalizable object?